packages/ra-core/src/controller/useReference.ts

```diff
@@ -0,0 +1,76 @@
+import { useEffect, useMemo } from 'react';
+// @ts-ignore
+import { useDispatch, useSelector } from 'react-redux';
+
+import { crudGetManyAccumulate } from '../actions';
+import { Record } from '../types';
+import { getReferenceResource } from '../reducer';
+
+interface Option {
+    id: string;
+    reference: string;
+    allowEmpty?: boolean;
+}
+
+export interface UseReferenceProps {
+    isLoading: boolean;
+    referenceRecord: Record;
+}
+
+/**
+ * @typedef ReferenceProps
+ * @type {Object}
+ * @property {boolean} isLoading: boolean indicating if the reference has loaded
+ * @property {Object} referenceRecord: the referenced record.
+ */
+
+/**
+ * Fetch reference record, and return it when avaliable
```

Comment:
I don't understand, this is the exact same jsDoc as `useReferenceField`. what's the difference between the two?